src/cprt_gridmap_filters/src/PreserveCostInflationFilter.cpp

```diff
@@ -0,0 +1,129 @@
+
+  if (!param_reader.get(std::string("input_layer"), this->inputLayer_)) {
+    RCLCPP_ERROR(
+        this->logging_interface_->get_logger(),
+        "PreserveCostInflationFilter did not find parameter 'input_layer'.");
+    return false;
+  }
+
+  if (!param_reader.get(std::string("output_layer"), this->outputLayer_)) {
+    RCLCPP_ERROR(
+        this->logging_interface_->get_logger(),
+        "PreserveCostInflationFilter did not find parameter 'output_layer'.");
+    return false;
+  }
+
+  if (!param_reader.get(std::string("inflation_radius"),
+                        this->inflationRadius_)) {
+    RCLCPP_ERROR(this->logging_interface_->get_logger(),
+                 "PreserveCostInflationFilter did not find parameter "
+                 "'inflation_radius'.");
+    return false;
+  }
+
+  return true;
+}
+
+template <typename T>
+bool PreserveCostInflationFilter<T>::update(const T &mapIn, T &mapOut) {
+  mapOut = mapIn;
+
+  // Check if layer exists.
+  if (!mapOut.exists(this->inputLayer_)) {
+    RCLCPP_ERROR(this->logging_interface_->get_logger(),
+                 "Layer %s does not exist. Unable to apply preserve cost "
+                 "inflation filter.",
+                 this->inputLayer_.c_str());
+    return false;
+  }
+
+  mapOut.add(this->outputLayer_, 1.0);
+
+  this->computeWithSimpleSerialMethod(mapIn, mapOut);
+
+  return true;
+}
+
+template <typename T>
+void PreserveCostInflationFilter<T>::computeWithSimpleSerialMethod(
+    const grid_map::GridMap &mapIn, grid_map::GridMap &mapOut) {
+  rclcpp::Clock clock;
+  const double start = clock.now().seconds();
+
+  const Eigen::MatrixXf &layerIn = mapOut[this->inputLayer_];
+
+  for (grid_map::GridMapIterator iterator(mapIn); !iterator.isPastEnd();
+       ++iterator) {
+    if (!mapIn.isValid(*iterator, this->inputLayer_)) {
+      continue;
+    }
+
+    const grid_map::Index index = *iterator;
+    grid_map::Position position;
+    mapIn.getPosition(
+        index, position);  // Get position of cell from grid_map::Index of cell
+
+    this->radialInflateSerial(mapOut, position,
+                              layerIn.coeff(index(0), index(1)));
+  }
+
+  const double end = clock.now().seconds();
+  RCLCPP_DEBUG_THROTTLE(this->logging_interface_->get_logger(), clock, 2.0,
+                        "NORMAL COMPUTATION TIME = %f", (end - start));
+}
+
+template <typename T>
+void PreserveCostInflationFilter<T>::radialInflateSerial(
+    grid_map::GridMap &mapOut, const grid_map::Position &position,
+    const float value) {
+  Eigen::MatrixXf &layerOut = mapOut[this->outputLayer_];
+
+  for (grid_map::CircleIterator iterator(mapOut, position,
+                                         this->inflationRadius_);
+       !iterator.isPastEnd(); ++iterator) {
+    if (!mapOut.isValid(*iterator, this->outputLayer_)) {
+      continue;
+    }
+
+    layerOut.coeffRef((*iterator)(0), (*iterator)(1)) =
+        std::min(layerOut.coeff((*iterator)(0), (*iterator)(1)), value);
```

Comment:
Does this work? It might be slightly more effiecient. Will need to include cmath
```suggestion
    const std::size_t index = iterator.getLinearIndex();
    auto & cellValue = layerOut(i);
    if (!std::isfinite(cellValue)){
      continue;
    }
    cellValue = std::min(cellValue, value);
```

Reply:
> Does this work? It might be slightly more effiecient. Will need to include cmath

The circle iterator doesn't have a getLinearIndex, and the layerOut(i) returns a Eigen::IndexedView which isn't compiling with the next function calls.

But I've updated it to use the reference to the cell value using the at function which is what they show in this demo file: https://github.com/ANYbotics/grid_map/blob/be3bbc9afbcebd6a5cadcb091211916a79d3dc14/grid_map_demos/src/IteratorsDemo.cpp#L91